atc/wrappa/pool.go

```diff
@@ -0,0 +1,37 @@
+package wrappa
+
+import (
+	"errors"
+	"sync"
+)
+
+type Pool interface {
+	TryAcquire() bool
+	Release() error
+}
+
+type pool struct {
+	size  int
+	state int
```

Comment:
obviously `uint64` is more annoying to write, but it is a more descriptive type...

Reply:
this seems a lot like a semaphore where we just don't have to bother about notifying waiters :thinking:  was there a specific reason not to use https://godoc.org/golang.org/x/sync/semaphore , I guess, aside that?

Reply:
I made almost exactly the same comment in fe8ee20b47a8eba3a62 😁 -- this implementation is simpler, doesn't introduce a dependency, and returns an error instead of panicking when you try to release the semaphore when it is not held.

Reply:
makes sense! and awesome to see that we're keeping that as part of the history :raised_hands: (and shame on me for not looking at the commit messages :sweat_smile: )

Reply:
No worries - I almost wonder if it's worth making a habit of sticking all the commit messages in the PR description so that you don't have to be vigilant about that. Otherwise your question is still valid - unless there's a strong reason for "gracefully" handling the "attempting to release an empty pool" case, it might be worth switching to the existing semaphore implementation because a) we already have a dependency on the library and b) it could allow removing a lot of code...

Reply:
We switched to the existing semaphore implementation!